crates/sui-graphql-rpc/src/types/big_int.rs

```diff
@@ -1,19 +1,38 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
+use std::str::FromStr;
+
 use async_graphql::*;
-use move_core_types::u256::U256;
 
-struct BigInt(U256);
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub(crate) struct BigInt(String);
 
-/// TODO: implement BigInt scalar type using u256
 #[Scalar]
 impl ScalarType for BigInt {
-    fn parse(_value: Value) -> InputValueResult<Self> {
-        unimplemented!()
+    fn parse(value: Value) -> InputValueResult<Self> {
+        match value {
+            Value::String(s) => {
+                // check that all are digits
+                if s.chars().all(|c| c.is_ascii_digit()) {
```

Comment:
`BigInt` can be signed, so we need to be able to support a leading `-` sign character.